sdk/lib/opentelemetry/sdk/resources/resource.rb

```diff
@@ -21,9 +21,10 @@ class << self
           # @return [Resource]
           def create(labels = {})
             frozen_labels = labels.each_with_object({}) do |(k, v), memo|
-              raise ArgumentError, 'label keys and values must be strings' unless k.is_a?(String) && v.is_a?(String)
+              raise ArgumentError, 'label keys must be strings' unless k.is_a?(String)
+              raise ArgumentError, 'label values must be strings, integers, floats, or booleans' unless Internal.valid_value?(v)
```

Comment:
This requires a documentation change as well (the `@param` in the comment above).